### ISSUE TITLE
Unresolved variables from Collection (not environment)

### ISSUE THREAD

**Issue by afilp** [a85/POSTMan-Chrome-Extension#4222]
When using `Variables` in Collection (not environment), we get `unresovled`:

![image](https://user-images.githubusercontent.com/7850073/36476184-cc8c5b4a-1705-11e8-8eb7-2035bea724ad.png)

![image](https://user-images.githubusercontent.com/7850073/36476187-ceb1944e-1705-11e8-96f6-c6cb1aa9f657.png)

## App Details:

```
Postman for Windows 10
v5.5.3
```

**Comment by callmeaatrey** [comment]
Thanks for reporting the issue, @afilp . It's already fixed and will be available to you in the upcoming release. Also, this issue is already addressed under #4105 , so we'll keep you posted on this there.

**Comment by afilp** [comment]
Thanks!

**Comment by qa-apps** [comment]
no its not fixed yet, it's almost 2020 and i still have this 'unresolved variable' issue.

**Comment by MuhammadimYusoff** [comment]
@callmeaatrey It is not fixed on Mac release. This below is the screenshot:-

1) I setup it i n my Collection Section:
<img width="636" alt="Screenshot 2020-02-04 at 3 27 58 PM" src="https://user-images.githubusercontent.com/43903543/73723296-ab548980-4763-11ea-9b31-79bf86914345.png">


2) I tested it on my GET request but "Unresolved Variable":
<img width="416" alt="Screenshot 2020-02-04 at 3 31 06 PM" src="https://user-images.githubusercontent.com/43903543/73723311-b1e30100-4763-11ea-9c96-cdd253f115a9.png">


3) This is my current Postman version release on MacOS:
<img width="720" alt="Screenshot 2020-02-04 at 3 28 38 PM" src="https://user-images.githubusercontent.com/43903543/73723372-db9c2800-4763-11ea-93d2-f41307caace5.png">


and it is not fixed, like period from 2018?

**Comment by ruslantagirov** [comment]
Same issue. I've created a variable in collection, but it remains unresolved.
Tried saving the request, closing, reopening, reloading Postman, etc.

Postman  v7.19.1, Windows 10 64 bit

**Comment by Stone624** [comment]
Bump.
It's September 2020 and this is a bug from February 2018.

I used to use Postman for my previous job and found it wonderful for development. Doing some heavy API calls and thought I'd return to this for the first time in 2 years. Looks like it's progressed a lot, But It's unfortunately that this is literally one of the first things I ran into upon downloading the application for MacOS Catalina. My experience is exactly as @MuhammadimYusoff . 

> Download and Login.
> Create a new workspace and collection.
> Set API credentials as collection variables because that sounds very helpful and convenient.
> Create a test API call to the server I'm working with.
> Unresolved variable despite the fact that I can click "Edit collection > variables" and see it.

Clearly the feature itself isn't necessary because you can use the request variables, but man, I wouldn't expect an application this well put together and old to have this significant of a bug as a first impression.

**Comment by nutmix** [comment]
its Sept 2020, and still this issue is not fixed?  I cant use postmans environment variables in some rest api collections, it always says "unresolved variable" for variables which are correctly defined in the current active env.  I have to hard code everything, and either use duplicates of the apis for different envs, or hand edit each api depending on the env.  Does anyone know of any work arounds to this issue?

I'm using 7.27.1 for mac.  It has worked in the past, but doesnt work currently.

Observations:

1.  some api collections work, some dont.  There is no difference between them.
2. When vars dont work, its always in the url, such as POST https://{{myhost}}/something.  myhost in this case is "undefined", even though its defined in the currently selected env, AND OTHER VARS IN THE BODY are picked up currectly... 


In console is this:
Error: getaddrinfo ENOTFOUND {{myhost}}

**Comment by rajshakerp** [comment]
My Environment was corrupted and Fixed this issue by duplicating an existing environment other than the corrupted environment ! Corrupted environment is not recoverable ! and changed the variables !

**Comment by WoodenCode00** [comment]
I had this issue when there was no environment choosen in Postman (pm.environment.set("my-var", pm.response.json().myProp;). I just choosen any env from the list of configured environments, and it worked. Now my variable is getting populated.

**Comment by rohithpr** [comment]
I ran into this issue, and it was caused by not clicking on "Save" after creating the variable.

Notice that the "Save" button is enabled indicating that there are some unsaved changes. This causes the "Unresolved variable" error:

[![enter image description here][1]][1]
[![enter image description here][2]][2]

After saving:
[![enter image description here][3]][3]
[![enter image description here][4]][4]


  [1]: https://i.stack.imgur.com/n1AeL.png
  [2]: https://i.stack.imgur.com/0BMLe.png
  [3]: https://i.stack.imgur.com/aK7OR.png
  [4]: https://i.stack.imgur.com/d7QDn.png

**Comment by pokehuy** [comment]
i saved everywhere i could, but it doesn't work until now...

**Comment by sari1fatih** [comment]
![1](https://user-images.githubusercontent.com/33065479/121779586-283c1280-cba5-11eb-989b-201420749578.png)

**Comment by PotatoesFall** [comment]
got this issue on v8.9.1 on Windows 10.
created a collection, added a request and referenced a variable in the url.
then added this variable to the collection (saved it and everything), but it was not recognized in the request.
workaround was to delete collection and add it again, this time adding the variable before referencing it in the URL.

**Comment by oyeraghib** [comment]
Saving it, solved the problem.

**Comment by qb20nh** [comment]
Make sure variables are declared without double braces
ex. `api-key`, not `{{api-key}}`

**Comment by YevheniiPokhvalii** [comment]
> I ran into this issue, and it was caused by not clicking on "Save" after creating the variable.
> 
> Notice that the "Save" button is enabled indicating that there are some unsaved changes. This causes the "Unresolved variable" error:
> 
> [![enter image description here](https://camo.githubusercontent.com/0b1925110e6b48fdcb3ac6dcdfee0714eabb8349c0d115bb4555ce06016e3ee0/68747470733a2f2f692e737461636b2e696d6775722e636f6d2f6e3141654c2e706e67)](https://i.stack.imgur.com/n1AeL.png) [![enter image description here](https://camo.githubusercontent.com/5727a0c0cec63dbb5908d079b35b04710661873b94e4169fe85450e7f7e9fb92/68747470733a2f2f692e737461636b2e696d6775722e636f6d2f30424d4c652e706e67)](https://i.stack.imgur.com/0BMLe.png)
> 
> After saving: [![enter image description here](https://camo.githubusercontent.com/6915fbc20244618adc561e070ea9dde7ccdb3266fbc2b46c58a1d63e204e0b73/68747470733a2f2f692e737461636b2e696d6775722e636f6d2f614b374f522e706e67)](https://i.stack.imgur.com/aK7OR.png) [![enter image description here](https://camo.githubusercontent.com/b5da0853e745b861c411b80d953e1d882b5e046a09e8827f9b20d4106703fb1c/68747470733a2f2f692e737461636b2e696d6775722e636f6d2f643751446e2e706e67)](https://i.stack.imgur.com/d7QDn.png)

It is not intuitive at all for the newcomers to Postman. I wish I had found your reply earlier - it took a lot of time to find out why it did not work. Yet, so simple solution - just click the "Save" button since it does not save automatically.

**Comment by malvikach** [comment]
@YevheniiPokhvalii I am sorry about your experience. I agree that there is a gap in the experience. This could be very difficult to spot for new users as you mentioned in your comment. Thanks for describing this issue so well and taking time to write your thoughts on this. We really appreciate it. 

We are tracking the issues related to this here[https://github.com/postmanlabs/postman-app-support/issues/8195](url)

On a sidenote- I would love to speak with you to get to know more about your experience as you have started using postman. It will help me identify more related problems. We are also looking at the experience of unresolved variables very closely and would be super helpful to get your feedback at this stage.

**Comment by ghost** [comment]
I disagree this is related to issue 8195 (which is about autosaving). This is a matter of inconsistent behavior of Collection Variables vs Environment Variables. This becomes very apparent when you only have read-only access to a Collection or Workspace.
* Even with read-only access, I can enter a "Current value" on a Environment Variable, and it comes into effect without saving.
* If I enter a "Current value" on a Collection Variable without saving, it does NOT come into effect. I believe it should. 

In summary, I think this deserves it's own issue.
